### PR TITLE
Added string to lower for port protocol

### PR DESCRIPTION
**Pull request by drewkerrigan**
Addresses DCOS-12154.

### CODE REVIEW

File: src/mesos_state_client.erl [hunk 1]
```diff
@@ -414,7 +414,7 @@ discovery_ports([Port = #{number := Number, protocol := Protocol}|RestPorts],
     PortRecord = #mesos_port{
        name = maps:get(name, Port, undefined),
        number = Number,
-       protocol = protocol(Protocol),
+       protocol = protocol(to_lower(Protocol)),
```
Comment (nlsun) [inline src/mesos_state_client.erl]: If you combine the `protocol()` and `to_lower()` functions then you can save the last `list_to_binary()` conversion

File: src/mesos_state_client.erl [hunk 2]
```diff
@@ -430,9 +430,12 @@ ports_from_discovery(#{ports := #{ports := Ports}}) ->
...
   [].
 
-protocol(<<"tcp">>) -> tcp;
-protocol(<<"udp">>) -> udp.
-
+protocol(BinString) when is_binary(BinString) ->
+    String = binary_to_list(BinString),
+    case string:to_lower(String) of
+        "tcp" -> tcp;
+        "udp" -> udp
```
Comment (aeyakovenko) [inline src/mesos_state_client.erl]: missing coverage for this case

Comment (drewkerrigan) [inline src/mesos_state_client.erl]: Added another port to state7.json for this.

**Comment by drewkerrigan** [comment]
Commits squashed, test case added for UDP

**Pull request merged**